engineapi/proxy.go

```diff
@@ -80,17 +83,62 @@ func NewEngineClientProxy(im *longhorn.InstanceManager, logger logrus.FieldLogge
 		return nil, err
 	}
 
-	ctx, cancel := context.WithCancel(context.Background())
-	client, err := imclient.NewProxyClient(ctx, cancel, im.Status.IP, InstanceManagerProxyServiceDefaultPort)
+	initProxyTLSClient := func(ip string) (*imclient.ProxyClient, error) {
+		// check for tls cert file presence
+		ctx, cancel := context.WithCancel(context.Background())
+		proxyClient, err := imclient.NewProxyClientWithTLS(ctx,
+			cancel,
+			ip,
+			InstanceManagerProxyServiceDefaultPort,
+			filepath.Join(types.TLSDirectoryInContainer, types.TLSCAFile),
+			filepath.Join(types.TLSDirectoryInContainer, types.TLSCertFile),
+			filepath.Join(types.TLSDirectoryInContainer, types.TLSKeyFile),
+			"longhorn-backend.longhorn-system",
+		)
+		defer func() {
+			if err != nil && proxyClient != nil {
+				proxyClient.Close()
+			}
+		}()
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to load Instance Manager Proxy Client TLS files")
+		}
+		if err = proxyClient.CheckConnection(); err != nil {
+			return nil, errors.Wrap(err, "failed to check Instance Manager Proxy Client with TLS connection")
+		}
+
+		return proxyClient, nil
+	}
+
+	proxyClient, err := initProxyTLSClient(im.Status.IP)
 	if err != nil {
-		return nil, err
+		logrus.WithError(err).Tracef("Falling back to non-tls client for Proxy Service Client for %v IP %v",
+			im.Name, im.Status.IP)
+		// fallback to non tls client, there is no way to differentiate between im versions unless we get the version via the im client
+		// TODO: remove this im client fallback mechanism in a future version maybe 2.4 / 2.5 or the next time we update the api version
+		ctx, cancel := context.WithCancel(context.Background())
+		proxyClient, err = imclient.NewProxyClient(ctx, cancel, im.Status.IP, InstanceManagerProxyServiceDefaultPort, nil)
+		defer func() {
+			if err != nil && proxyClient != nil {
+				proxyClient.Close()
+			}
+		}()
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to initialize Proxy Service Client for %v IP %v",
+				im.Name, im.Status.IP)
+		}
+
+		if err = proxyClient.CheckConnection(); err != nil {
+			return nil, errors.Wrapf(err, "failed to check Proxy Service Client connection for %v IP %v",
+				im.Name, im.Status.IP)
+		}
```

Comment:
I just mirrored the similar methods here (as @sfackler did). Maybe we can refactor them all after the v1.6.0 release.